rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -2480,6 +2481,30 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
      * evaluate the options in the input argString, and if any Options defined for the Job have required=true, have a
      * defaultValue, and have null value in the input properties, then append the default option value to the argString
      */
+    def Map addRemoteOptionSelected(ScheduledExecution scheduledExecution, Map optparams, UserAndRolesAuthContext authContext = null) throws ExecutionServiceException {
+        def newmap = new HashMap(optparams)
+
+        final options = scheduledExecution.options
+        if (options) {
+            def defaultoptions=[:]
+            options.each {Option opt ->
+                if(null==optparams[opt.name] && opt.enforced && !opt.optionValues){
+                    Map remoteOptions = scheduledExecutionService.loadOptionsRemoteValues(scheduledExecution, [option: opt.name], authContext?.username)
```

Comment:
I think the `mapConfig` parameter should also include the `optparams` in case the url has references to other options, like this:
```suggestion
                    Map remoteOptions = scheduledExecutionService.loadOptionsRemoteValues(scheduledExecution, [option: opt.name, extra: [option: optparams]], authContext?.username)
```